nextjs-13/src/app/api/v1/newsletter/[id]/route.ts

```diff
@@ -0,0 +1,33 @@
+import { StatusCodes } from 'http-status-codes';
+import { NextResponse, NextRequest } from 'next/server';
+
+import appHandler from 'lib/handler/app.handler';
+import { deleteNewsletter } from 'repositories/newsletter.repository';
+
+export async function DELETE(
+  req: NextRequest,
+  { params }: { params: { id: string } }
+) {
+  return appHandler(req, async (currentUser) => {
+    try {
+      const result = await deleteNewsletter(params.id, currentUser.id);
+
+      if (result.count === 0) {
+        return NextResponse.json(
+          { message: 'Newsletter not exists' },
```

Comment:
Minor english syntax issue: 

```suggestion
          { message: 'Newsletter does not exist' },
```

But we could be more generic:

```suggestion
          { message: 'Newsletter could not be deleted' },
```

Reply:
Ahh my bad my bad 🙈 I do use Grammarly, but just for long writing text, not for the code much. I remember to check test names but forget checking response messages this time 🙈

Reply:
Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/113/commits)